### DISCUSSION TITLE
Can I use ops and nanos to compile and run untrusted code?

### DISCUSSION THREAD

**Discussion by kerolloz** [nanovms/ops#1180] [General]
Is it safe to use _nanos_ and _ops_ to compile and run probably-unsafe source code (e.g.: use nanos to run code for an online code editor)?

**Comment by francescolavra** [comment]
It depends on what the untrusted code would be doing behind the scenes.
If it tries to exploit known OS vulnerabilities to gain access to data from other applications running in the system or execute third party applications, then running it on Nanos is certainly safer than using a general purpose OS such as Linux, because in Nanos there are no other running applications and it's not possible to start new ones.
If on the other end the untrusted code does things that only need to access your application's data (for example opens a backdoor that allows retrieving your application's memory contents via the network), then there is no OS (unikernel or otherwise) that can prevent this kind of things; you could scan the created executable with an antivirus, but that won't give you absolute certainty that there is no malicious code in it.
